### PR TITLE
Register cluster for active health check when it's enabled on reload

### LINKED ISSUE #1134: Active health checks are not enabled after configuration reload
### Describe the bug
Active health checks are not enabled after configuration reload, only after a full restart. Strange thing is that the disabling seems to work, only the enabling doesn't

### To Reproduce

1. Create a config with a cluster with an active health check, but disabled (Enabled = false)
2. Start the proxy
3. While running the proxy, change and save the config, setting the health check to enabled=true
4. Watch the reload of the config happening, but the health checks calls don't appear in the logs

### Further technical details

- Yarp preview 12
- Windows

### PR DESCRIPTION
**Pull request by alnikola**
Fixes #1134

### CODE REVIEW

File: src/ReverseProxy/Health/ActiveHealthCheckMonitor.cs [hunk 1]
```diff
@@ -83,24 +81,26 @@ public void OnClusterChanged(ClusterState cluster)
...
+            Scheduler.UnscheduleEntity(cluster);
         }
 
         public void Dispose()
         {
-            _scheduler.Dispose();
+            Scheduler.Dispose();
         }
 
+        internal EntityActionScheduler<ClusterState> Scheduler { get; private set; }
```
Comment (Kahbazi) [inline src/ReverseProxy/Health/ActiveHealthCheckMonitor.cs:102]: ?
```suggestion
        internal EntityActionScheduler<ClusterState> Scheduler { get; }
```

Comment (alnikola) [reply src/ReverseProxy/Health/ActiveHealthCheckMonitor.cs:102]: Thanks!

File: src/ReverseProxy/Health/ActiveHealthCheckMonitor.cs [hunk 2]
```diff
@@ -83,24 +81,26 @@ public void OnClusterChanged(ClusterState cluster)
...
+            Scheduler.UnscheduleEntity(cluster);
         }
 
         public void Dispose()
         {
-            _scheduler.Dispose();
+            Scheduler.Dispose();
         }
 
+        internal EntityActionScheduler<ClusterState> Scheduler { get; }
```
Comment (Tratcher) [inline src/ReverseProxy/Health/ActiveHealthCheckMonitor.cs:102]: Organization: properties go before methods, even internal ones.

**Review by Tratcher — APPROVED** [review]

File: src/ReverseProxy/Health/ActiveHealthCheckMonitor.cs (hunk 2 above)
Comment (alnikola) [reply src/ReverseProxy/Health/ActiveHealthCheckMonitor.cs]: Fixed

**Pull request merged**